src/db.rs

```diff
@@ -16,13 +15,16 @@ use crate::sst::SsTableFormat;
 use crate::sst_iter::SstIterator;
 use crate::tablestore::TableStore;
 use crate::types::ValueDeletable;
+use crate::{block::Block, compactor::Compactor};
 use bytes::Bytes;
 use fail_parallel::FailPointRegistry;
 use object_store::path::Path;
 use object_store::ObjectStore;
 use parking_lot::{Mutex, RwLock};
 use tokio::runtime::Handle;
 
+pub type BlockCache = moka::future::Cache<(SsTableId, usize), Arc<Block>>;
```

Comment:
Couple thoughts on the data type here:

I think it's important that we be able to use the cache to cache index and bloom filter "blocks" as well as data blocks. These will actually likely be the most frequently accessed parts of a given SST, so ideally we're getting them from cache. Have you thought about how we could support this? One idea would be to use a more broad key and value type like:
```
enum CachedBlock {
    Data(Arc<Block>),
    Index(Arc<SsTableIndexOwned>),
    Filter(Arc<BloomFilter>),
}
```

Also, I think it would be good to put the moka cache behind some trait so we're not baking in that implementation detail into the db/tablestore.

Reply:
Yeah I couldn't figure out how we could fit in a trait, but the enum worked.